### PR TITLE
add k3s flavor support

### PR DESCRIPTION
**Pull request by eljohnson92**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR adds template support for k3s as a flavor when generating a cluster. it does require that the k3s bootstrap/controlplane providers are already installed and also that you specify a k3s specific k8s version

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: templates/flavors/k3s/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      region: ${LINODE_REGION}
+  cilium.yaml: |-
+    apiVersion: helm.cattle.io/v1
+    kind: HelmChart
+    metadata:
+       namespace: kube-system
+       name: cilium
+    spec:
+       targetNamespace: kube-system
+       version: v1.15.0
```
Comment (AshleyDumaine) [inline templates/flavors/k3s/secret.yaml:24]: can we have this version be a variable? https://github.com/linode/cluster-api-provider-linode/blob/main/templates/addons/cilium/cilium.yaml#L12

File: templates/flavors/k3s/secret.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+             enabled: true
+  linode-ccm.yaml: |-
+    apiVersion: helm.cattle.io/v1
+    kind: HelmChart
+    metadata:
+     namespace: kube-system
+     name: ccm-linode
+    spec:
+     targetNamespace: kube-system
+     version: v0.3.24
```
Comment (AshleyDumaine) [inline templates/flavors/k3s/secret.yaml:42]: same here

File: templates/flavors/k3s/secret.yaml
Comment (AshleyDumaine) [inline templates/flavors/k3s/secret.yaml:1]: Out of scope of this PR but I wonder if we should have this for the default (kubeadm) flavor as well

File: docs/src/flavors/flavors.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+```bash
+clusterctl generate cluster test-cluster --flavor clusterclass
+```
+
+To use the default flavor, omit the `--flavor` flag.
+
+See the [`clusterctl` flavors docs](https://cluster-api.sigs.k8s.io/clusterctl/commands/generate-cluster.html#flavors) for more information.
+
+
+## supported flavors
```
Comment (AshleyDumaine) [inline docs/src/flavors/flavors.md:16]:
```suggestion
## Supported flavors 
```

File: docs/src/topics/getting-started.md [hunk 4]
```diff
@@ -17,14 +17,22 @@ For more information please see the
 [Linode Guide](https://www.linode.com/docs/products/tools/api/guides/manage-api-tokens/#create-an-api-token).
 ```
 
-## Setting up your Linode environment
+## Setting up your cluster environment variable
```
Comment (AshleyDumaine) [inline docs/src/topics/getting-started.md:20]:
```suggestion
## Setting up your cluster environment variables
```

File: docs/src/topics/getting-started.md [hunk 5]
```diff
@@ -17,14 +17,22 @@ For more information please see the
...
+export LINODE_MACHINE_TYPE=g6-standard-2
 ```
 
-## Building your first cluster
 
-Please continue from the [setting up the environment](../developers/development.md#setting-up-the-environment)
-section for creating your first Kubernetes cluster on Linode using Cluster API. 
+## Deploying your first cluster
+
+please refer to the [default flavor](../flavors/default.md) section for creating your first Kubernetes cluster on Linode using Cluster API.
```
Comment (AshleyDumaine) [inline docs/src/topics/getting-started.md:38]:
```suggestion
Please refer to the [default flavor](../flavors/default.md) section for creating your first Kubernetes cluster on Linode using Cluster API. 
```

File: docs/src/flavors/clusterclass-kubeadm.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
+# Kubeadm ClusterClass
+## Specification
+| Control Plane | CNI    | OS     | Installs ClusterClass |
+|---------------|--------|--------|-----------------------|
+| Kubeadm       | Cilium | Ubuntu | Yes                   | 
+## Prerequisites
+[Quickstart](../topics/getting-started.md) completed
+## Usage
+### Create clusterClass and first cluster
+1. Generate clusterClass & cluster manifests
```
Comment (AshleyDumaine) [inline docs/src/flavors/clusterclass-kubeadm.md:10]:
```suggestion
1. Generate the ClusterClass and cluster manifests:
```

File: docs/src/flavors/clusterclass-kubeadm.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    ```bash
+    clusterctl generate cluster test-cluster --flavor clusterclass-kubeadm > test-cluster.yaml
+    ```
+2. Apply cluster manifests
+    ```bash
+    kubectl apply -f test-cluster.yaml
+    ```
+### Create second cluster
```
Comment (AshleyDumaine) [inline docs/src/flavors/clusterclass-kubeadm.md:18]: I would note that this is optional, e.g. "you can create a second cluster with the existing ClusterClass as follows"

**Review by mhmxs — APPROVED** [review]

File: docs/src/flavors/k3s.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  export KUBERNETES_VERSION=v1.29.1+k3s2
+  ```
+* Installed [k3s bootstrap provider](https://github.com/k3s-io/cluster-api-k3s) into your management cluster
+  * Add `k3s-clusterctl.yaml` config for the k3s bootstrap/control plane providers
+    ```yaml
+    providers:
+      - name: "k3s"
+        url: https://github.com/k3s-io/cluster-api-k3s/releases/latest/bootstrap-components.yaml
+        type: "BootstrapProvider"
+      - name: "k3s"
```
Comment (AshleyDumaine) [inline docs/src/flavors/k3s.md:19]: Are we missing a type here?
```
Fetching providers
Error: failed to get provider components for the "k3s" provider: error validating configuration for the  with name k3s. Please fix the providers value in clusterctl configuration file: invalid provider type. Allowed values are [CoreProvider, BootstrapProvider, InfrastructureProvider, ControlPlaneProvider, IPAMProvider, RuntimeExtensionProvider, AddonProvider]
```

File: docs/src/flavors/k3s.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
+# K3s
+## Specification
+| Control Plane               | CNI    | OS     | Installs ClusterClass |
+|-----------------------------|--------|--------|-----------------------|
+| [k3s](https://docs.k3s.io/) | Cilium | Ubuntu | No                    |
+## Prerequisites
+* [Quickstart](../topics/getting-started.md) completed
+* Select a [k3s kubernetes version](https://github.com/k3s-io/k3s/releases) to set for the kubernetes version
+  ```bash
+  export KUBERNETES_VERSION=v1.29.1+k3s2
```
Comment (AshleyDumaine) [inline docs/src/flavors/k3s.md:10]:
```suggestion
  export K3S_KUBERNETES_VERSION=v1.29.1+k3s2
```

File: docs/src/flavors/flavors.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+```
+
+To use the default flavor, omit the `--flavor` flag.
+
+See the [`clusterctl` flavors docs](https://cluster-api.sigs.k8s.io/clusterctl/commands/generate-cluster.html#flavors) for more information.
+
+
+## Supported flavors 
+
+- [Default](default.md)
```
Comment (AshleyDumaine) [inline docs/src/flavors/flavors.md:18]:
```suggestion
- [Default (kubeadm)](default.md)
```

File: docs/src/flavors/clusterclass-kubeadm.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
+# Kubeadm ClusterClass
+## Specification
+| Control Plane | CNI    | OS     | Installs ClusterClass |
+|---------------|--------|--------|-----------------------|
+| Kubeadm       | Cilium | Ubuntu | Yes                   |
```
Comment (AshleyDumaine) [inline docs/src/flavors/clusterclass-kubeadm.md:5]:
```suggestion
| Control Plane | CNI    | Default OS   | Installs ClusterClass |
|---------------|--------|--------------|-----------------------|
| Kubeadm       | Cilium | Ubuntu 22.04 | Yes                   | 
```

**Review by AshleyDumaine — APPROVED** [review]
LGTM

**Pull request merged**